src/realtime_metrics/main.py

```diff
@@ -17,59 +18,69 @@ def run_analysis():
 
-            # add stop_time_update to dictionary if not present or update if timestamp of tripUpdate is newer than the one in the dictionary
+            # add stop time update to trips
             stop_time_update: StopTimeUpdate = trip_stop_time_update.StopTimeUpdate
             key = (tripUpdate.route_id, tripUpdate.trip_id, stop_time_update.stop_id)
-            if key in dictionary.keys():
-                if tripUpdate.timestamp.timestamp() > dictionary[key][0]:
-                    dictionary[key] = (tripUpdate.timestamp.timestamp(), stop_time_update)
+            if key in trips.keys():
+                trip_updates: list[TripUpdate] = trips[key]
             else:
-                dictionary[key] = (tripUpdate.timestamp.timestamp(), stop_time_update)
+                trip_updates: list[TripUpdate] = []
+            trip_updates.append((trip_stop_time_update.TripUpdate, trip_stop_time_update.StopTimeUpdate))
+            trips[key] = trip_updates
 
-            # add stop time update to trips_dict
-            if key in trips_dict.keys():
-                trip_list: list[TripUpdate] = trips_dict[key]
+            # add stop_time_update to dictionary if not present or update if timestamp of tripUpdate is newer than the one in the dictionary
+            if stop_time_update.arrival_uncertainty > 0:
+                continue # only consider updates with arrival uncertainty 0 as actual arrivals
+            if key in actual_arrival_times.keys():
+                if tripUpdate.timestamp.replace(tzinfo=timezone.utc).timestamp() > actual_arrival_times[key][0]:
+                    actual_arrival_times[key] = (tripUpdate.timestamp.replace(tzinfo=timezone.utc).timestamp(), stop_time_update)
             else:
-                trip_list: list[TripUpdate] = []
-            trip_list.append((trip_stop_time_update.TripUpdate, trip_stop_time_update.StopTimeUpdate))
-            trips_dict[key] = trip_list
+                actual_arrival_times[key] = (tripUpdate.timestamp.replace(tzinfo=timezone.utc).timestamp(), stop_time_update)
 
     mse_accuracy_result = mse_accuracy(stop_time_updates=stop_time_updates)
-    print("MSE accuracy: ", mse_accuracy_result)
+    if mse_accuracy_result is None:
+        print("MSE accuracy could not be computed, no data provided!")
+    else:
+        print("MSE accuracy: ", mse_accuracy_result)
+
     eta_accuracy_result = eta_accuracy(stop_time_updates=stop_time_updates)
-    print("ETA accuracy: ", eta_accuracy_result)
+    if eta_accuracy_result is None:
+        print("ETA accuracy could not be computed, no data provided!")
+    else:
+        print("ETA accuracy: ", eta_accuracy_result)
+
+    experienced_wait_time_delay_result = experienced_wait_time_delay(stop_time_updates)
+    if experienced_wait_time_delay_result is None:
+        print("Experienced Wait Time Delay could not be computed, no data provided!")
+    else:
+        print("Experienced Wait Time Delay: ", experienced_wait_time_delay_result)
```

Comment:
For the accuracy it is not needed, but Experienced wait time delay should give a unit, to know whether its in minutes or seconds or what